src/Iddict.elm

```diff
@@ -118,13 +118,16 @@ empty =
 -}
 fromList : List ( Int, a ) -> Iddict a
 fromList items =
+    let
+        ( cursor, dict ) =
+            List.foldl
```

Comment:
Note: the implementation of `Dict.fromList` already uses `List.foldl` in [miniBill/elm-fast-dict](https://github.com/miniBill/elm-fast-dict/blob/e381d089e55fbb7ac719f0e3e6f1470fe9dc8eef/src/FastDict.elm#L1032).

Folding two values together is an `O(n)` implementation, and folding them separately should come to the same complexity of `O(2n) = O(n)`. Does it pay off to do `List.foldl` ourselves?

Reply:
The big-O complexity is - in fact - the same, but the difference is that instead of traversing through the list three times (`Dict.fromList`, `List.map`, `List.maximum`), and also building an intermediate list (the `List.map` part), we just pass through it once - so we don't even need additional memory and it should be a bit faster - which matters for a library (it probably would be fine for application code if you know your dictionaries are always going to be small, but for libraries it's a good idea to optimize for all users).

You are correct that a simple `(+) 1` would fix the bug. Do you want me to separate that into another PR so you can merge it while we discuss this one?

Reply:
No, your argument sounds reasonable. 👍 I am marking this as resolved!